src/nrniv/nrncore_write/callbacks/nrncore_callbacks.cpp

```diff
@@ -29,7 +29,10 @@ extern char* pnt_map;
 /** Populate function pointers by mapping function pointers for callback */
 void map_coreneuron_callbacks(void* handle) {
     for (int i=0; cnbs[i].name; ++i) {
-        void* sym = dlsym(handle, cnbs[i].name);
+        void* sym = NULL;
+#if defined(HAVE_DLFCN_H)
+        sym = dlsym(handle, cnbs[i].name);
+#endif
         if (!sym) {
             fprintf(stderr, "Could not get symbol %s from CoreNEURON\n", cnbs[i].name);
```

Comment:
So how callbacks are setup on windows? This will always fail?

Reply:
On windows one has to in effect use the idiom in nrn/src/nrniv/nrnpy.cpp (also used in several other places)
```
#ifdef MINGW
#define RTLD_NOW 0
#define RTLD_GLOBAL 0
#define RTLD_NOLOAD 0
extern "C" {
extern void* dlopen_noerr(const char* name, int mode);
#define dlopen dlopen_noerr
extern void* dlsym(void* handle, const char* name);
extern int dlclose(void* handle); 
extern char* dlerror();
}
#else
//#define _GNU_SOURCE
#include <dlfcn.h>
#endif
```
As this problem was introduced in the nrnbbcore_write factorization, which also did not support windows, it seemed simplest to work around with this change. There are two other places in the nrnbbcore_write factorization which already
use the ```#if defined(HAVE_DLFCN_H)``` idiom. When work begins on making CoreNEURON available in windows, I think it would make sense to gather all these ```#ifdef MINGW``` fragments into an include file. But a bit of a review will be needed to understand the role of the mac specific nrn/src/nrnoc/osxdlfcn.h file. That is used in nrnoc/init.c probably for a no longer valid legacy reason. I note that with this pull request, the azure nrn-nightly-AMD64.exe artifact still does not have a complete neuron module. I was hoping this would fix that. I'm puzzled how the azure build was ever able to compile without error in the absence of this fix. I wonder if the latest mingw/msys2 has a dlfcn.h? I'll upgrade my msys2 and look into that.

Reply:
I have a bunch of those incoming in my CPP PR , https://github.com/neuronsimulator/nrn/pull/763/commits/d1d43e003860dc592e71c9df6ba0cc06fad5d61f